From 93bf2b2fed63dcc683cdbba4e3af7b9f45c5aba3 Mon Sep 17 00:00:00 2001
From: jfriedt <jean-michel.friedt@femto-st.fr>
Date: Tue, 23 Feb 2021 11:09:19 +0100
Subject: [PATCH] Makefile: static and dynamic library generation + install

Signed-off-by: Jean-Michel Friedt <friedtj@free.fr>
---
 Makefile | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index a414ce7..b9547b9 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
-CXXFLAGS = -std=c++14 -I.
+CXXFLAGS = -std=c++14 -I. -fPIC
 LDFLAGS = -lm
 
-TARGETS = gen_ft8 decode_ft8 test
+TARGETS = gen_ft8 decode_ft8 test libft8.so libft8.a
 
 .PHONY: run_tests all clean
 
@@ -19,8 +19,15 @@ test:  test.o ft8/pack.o ft8/encode.o ft8/text.o ft8/constants.o fft/kiss_fftr.o
 decode_ft8: decode_ft8.o fft/kiss_fftr.o fft/kiss_fft.o ft8/decode.o ft8/encode.o ft8/ldpc.o ft8/unpack.o ft8/text.o ft8/constants.o common/wave.o
 	$(CXX) $(LDFLAGS) -o $@ $^
 
+libft8.a: ft8/constants.o ft8/encode.o ft8/pack.o ft8/text.o common/wave.o
+	$(AR) rc libft8.a ft8/constants.o ft8/encode.o ft8/pack.o ft8/text.o common/wave.o
+
+libft8.so: ft8/constants.o ft8/encode.o ft8/pack.o ft8/text.o common/wave.o
+	$(CXX) -shared -o $@ $^
+
 clean:
 	rm -f *.o ft8/*.o common/*.o fft/*.o $(TARGETS)
+
 install:
-	$(AR) rc libft8.a ft8/constants.o ft8/encode.o ft8/pack.o ft8/text.o common/wave.o 
 	install libft8.a /usr/lib/libft8.a
+	install libft8.so /usr/lib/libft8.so
-- 
2.30.0

