diff --git a/src/Makefile b/src/Makefile
index 78f26c7..854f17d 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,89 +1,89 @@
 all: ../pisstv ../piopera ../pifsq ../pichirp ../sendiq ../tune ../freedv ../pocsag ../spectrumpaint ../pifmrds ../rpitx ../corel8 ../pift8 ../sendook ../morse ../foxhunt
 
-CFLAGS	= -Wall -g -O2 -Wno-unused-variable
-CXXFLAGS = -std=c++11 -Wall -g -O2 -Wno-unused-variable
-LDFLAGS = librpitx/src/librpitx.a -lm -lrt -lpthread -L/opt/vc/lib -lbcm_host
-CCP = c++
-CC = cc
+CFLAGS	?= -Wall -g -O2 -Wno-unused-variable
+CXXFLAGS ?= -std=c++11 -Wall -g -O2 -Wno-unused-variable
+LDFLAGS = -lrpitx -lm -lrt -lpthread -L/opt/vc/lib 
+CXX ?= c++
+CC ?= cc
 
 CFLAGS_Pissb	= -Wall -g -O2 -Wno-unused-variable
-LDFLAGS_Pissb	= librpitx/src/librpitx.a -lm -lrt -lpthread -lsndfile -lliquid -L/opt/vc/lib -lbcm_host
+LDFLAGS_Pissb	= -lm -lrt -lpthread -lsndfile -lliquid -L/opt/vc/lib -lbcm_host -lrpitx
 
-../pissb: ssbgen/test_ssb.c ssbgen/ssb_gen.c ssbgen/liquid_ssb.c librpitx/src/librpitx.a
+../pissb: ssbgen/test_ssb.c ssbgen/ssb_gen.c ssbgen/liquid_ssb.c 
 	$(CC) $(CFLAGS_Pissb) -o ../pissb ssbgen/liquid_ssb.c $(LDFLAGS_Pissb)
 
-../pisstv : sstv/pisstv.cpp librpitx/src/librpitx.a
-	$(CCP) $(CXXFLAGS) -o ../pisstv sstv/pisstv.cpp  $(LDFLAGS)
+../pisstv : sstv/pisstv.cpp 
+	$(CXX) $(CXXFLAGS) -o ../pisstv sstv/pisstv.cpp  $(LDFLAGS)
 	
-../foxhunt : foxhunt/foxhunt.cpp librpitx/src/librpitx.a
-	$(CCP) $(CXXFLAGS) -o ../foxhunt foxhunt/foxhunt.cpp  $(LDFLAGS)
+../foxhunt : foxhunt/foxhunt.cpp 
+	$(CXX) $(CXXFLAGS) -o ../foxhunt foxhunt/foxhunt.cpp  $(LDFLAGS)
 
-../piopera : opera/opera.cpp librpitx/src/librpitx.a
-	$(CCP) $(CXXFLAGS) -o ../piopera opera/opera.cpp  $(LDFLAGS)
+../piopera : opera/opera.cpp 
+	$(CXX) $(CXXFLAGS) -o ../piopera opera/opera.cpp  $(LDFLAGS)
 
-../pifsq : fsq/pifsq.cpp librpitx/src/librpitx.a
-	$(CCP) $(CXXFLAGS) -o ../pifsq fsq/pifsq.cpp  $(LDFLAGS)
+../pifsq : fsq/pifsq.cpp 
+	$(CXX) $(CXXFLAGS) -o ../pifsq fsq/pifsq.cpp  $(LDFLAGS)
 
-../pichirp : chirp/chirp.cpp librpitx/src/librpitx.a
-	$(CCP) $(CXXFLAGS) -o ../pichirp chirp/chirp.cpp  $(LDFLAGS)
+../pichirp : chirp/chirp.cpp 
+	$(CXX) $(CXXFLAGS) -o ../pichirp chirp/chirp.cpp  $(LDFLAGS) 
 
-../morse : morse/morse.cpp librpitx/src/librpitx.a
-	$(CCP) $(CXXFLAGS) -o ../morse morse/morse.cpp  $(LDFLAGS)
+../morse : morse/morse.cpp 
+	$(CXX) $(CXXFLAGS) -o ../morse morse/morse.cpp  $(LDFLAGS)
 
-../sendiq : sendiq.cpp librpitx/src/librpitx.a
-	$(CCP) $(CXXFLAGS) -o ../sendiq sendiq.cpp  $(LDFLAGS)
+../sendiq : sendiq.cpp 
+	$(CXX) $(CXXFLAGS) -o ../sendiq sendiq.cpp  $(LDFLAGS)
 
-../tune : tune.cpp librpitx/src/librpitx.a
-	$(CCP) $(CXXFLAGS) -o ../tune tune.cpp  $(LDFLAGS)
+../tune : tune.cpp 
+	$(CXX) $(CXXFLAGS) -o ../tune tune.cpp  $(LDFLAGS)
 
-../freedv : freedv/freedv.cpp librpitx/src/librpitx.a
-	$(CCP) $(CXXFLAGS) -o ../freedv freedv/freedv.cpp  $(LDFLAGS)
+../freedv : freedv/freedv.cpp 
+	$(CXX) $(CXXFLAGS) -o ../freedv freedv/freedv.cpp  $(LDFLAGS)
 
-../dvbrf : dvb/dvbrf.cpp dvb/dvbsenco8.s dvb/fec100.c dvb/dvbs2arm_1v30.s librpitx/src/librpitx.a
+../dvbrf : dvb/dvbrf.cpp dvb/dvbsenco8.s dvb/fec100.c dvb/dvbs2arm_1v30.s 
 	$(CC) $(CFLAGS) -c -o dvb/dvbsenco8.o dvb/dvbsenco8.s
 	$(CC) $(CFLAGS) -c -o dvb/dvbs2arm_1v30.o dvb/dvbs2arm_1v30.s
 	$(CC) $(CFLAGS) -c -o dvb/fec100.o dvb/fec100.c
-	$(CCP) $(CXXFLAGS) -o ../dvbrf dvb/dvbrf.cpp dvb/dvbsenco8.o dvb/fec100.o dvb/dvbs2arm_1v30.o $(LDFLAGS)
+	$(CXX) $(CXXFLAGS) -o ../dvbrf dvb/dvbrf.cpp dvb/dvbsenco8.o dvb/fec100.o dvb/dvbs2arm_1v30.o $(LDFLAGS)
 
-../pocsag: pocsag/pocsag.cpp librpitx/src/librpitx.a
-	$(CCP) $(CXXFLAGS) -o ../pocsag pocsag/pocsag.cpp $(LDFLAGS)
+../pocsag: pocsag/pocsag.cpp 
+	$(CXX) $(CXXFLAGS) -o ../pocsag pocsag/pocsag.cpp $(LDFLAGS)
 
-../spectrumpaint: spectrumpaint/spectrum.cpp librpitx/src/librpitx.a
-	$(CCP) $(CXXFLAGS) -o ../spectrumpaint spectrumpaint/spectrum.cpp $(LDFLAGS)
+../spectrumpaint: spectrumpaint/spectrum.cpp 
+	$(CXX) $(CXXFLAGS) -o ../spectrumpaint spectrumpaint/spectrum.cpp $(LDFLAGS)
 
-../pifmrds: pifmrds/rds.c pifmrds/waveforms.c pifmrds/pi_fm_rds.cpp pifmrds/fm_mpx.c pifmrds/control_pipe.c librpitx/src/librpitx.a
+../pifmrds: pifmrds/rds.c pifmrds/waveforms.c pifmrds/pi_fm_rds.cpp pifmrds/fm_mpx.c pifmrds/control_pipe.c 
 	$(CC) $(CFLAGS) -c -o pifmrds/rds.o pifmrds/rds.c
 	$(CC) $(CFLAGS) -c -o pifmrds/control_pipe.o pifmrds/control_pipe.c
 	$(CC) $(CFLAGS) -c -o pifmrds/waveforms.o pifmrds/waveforms.c
 	$(CC) $(CFLAGS) -c -o pifmrds/rds_wav.o pifmrds/rds_wav.c
 	$(CC) $(CFLAGS) -c -o pifmrds/fm_mpx.o pifmrds/fm_mpx.c
 	$(CC) -o pifmrds/rds_wav pifmrds/rds_wav.o pifmrds/rds.o pifmrds/waveforms.o pifmrds/fm_mpx.o -lm -lsndfile
-	$(CCP) $(CXXFLAGS) -Wno-write-strings -o ../pifmrds pifmrds/rds.o pifmrds/waveforms.o pifmrds/pi_fm_rds.cpp pifmrds/fm_mpx.o pifmrds/control_pipe.o librpitx/src/librpitx.a -lm -lsndfile -lrt -lpthread -L/opt/vc/lib -lbcm_host
+	$(CXX) $(CXXFLAGS) -Wno-write-strings -o ../pifmrds pifmrds/rds.o pifmrds/waveforms.o pifmrds/pi_fm_rds.cpp pifmrds/fm_mpx.o pifmrds/control_pipe.o -lm -lsndfile -lrt -lpthread -L/opt/vc/lib -lrpitx
 
-../rpitx: rpitxv1/rpitx.cpp librpitx/src/librpitx.a
-	$(CCP) $(CXXFLAGS) -Wno-write-strings -o ../rpitx rpitxv1/rpitx.cpp $(LDFLAGS)
+../rpitx: rpitxv1/rpitx.cpp 
+	$(CXX) $(CXXFLAGS) -Wno-write-strings -o ../rpitx rpitxv1/rpitx.cpp $(LDFLAGS)
 
-../corel8: corel8/corel8.cpp corel8/costas8.h librpitx/src/librpitx.a
-	$(CCP) $(CXXFLAGS) -Wno-write-strings -o ../corel8 corel8/corel8.cpp $(LDFLAGS)
+../corel8: corel8/corel8.cpp corel8/costas8.h 
+	$(CXX) $(CXXFLAGS) -Wno-write-strings -o ../corel8 corel8/corel8.cpp $(LDFLAGS)
 
-../pift8 : pift8/pift8.cpp pift8/ft8_lib/ft8/constants.cpp pift8/ft8_lib/ft8/text.cpp pift8/ft8_lib/ft8/pack.cpp pift8/ft8_lib/ft8/encode.cpp librpitx/src/librpitx.a
-	$(CCP) $(CXXFLAGS) -Wno-write-strings -o ../pift8 pift8/pift8.cpp pift8/ft8_lib/ft8/constants.cpp pift8/ft8_lib/ft8/text.cpp pift8/ft8_lib/ft8/pack.cpp pift8/ft8_lib/ft8/encode.cpp  $(LDFLAGS)
+../pift8 : pift8/pift8.cpp  
+	$(CXX) $(CXXFLAGS) -Wno-write-strings -o ../pift8 pift8/pift8.cpp -lft8 $(LDFLAGS) 
 
-../sendook: ook/sendook.cpp ook/optparse.c librpitx/src/librpitx.a
-	$(CCP) $(CXXFLAGS) -Wno-write-strings -o ../sendook ook/sendook.cpp ook/optparse.c $(LDFLAGS)
+../sendook: ook/sendook.cpp ook/optparse.c 
+	$(CXX) $(CXXFLAGS) -Wno-write-strings -o ../sendook ook/sendook.cpp ook/optparse.c $(LDFLAGS)
 
 CFLAGS_Pifm	= -Wall -g -O2 -Wno-unused-variable
-LDFLAGS_Pifm	= librpitx/src/librpitx.a -lm -lrt -lpthread -lsndfile -L/opt/vc/lib -lbcm_host
+LDFLAGS_Pifm	= -lm -lrt -lpthread -lsndfile -L/opt/vc/lib -lrpitx
 ../pifm : ../fm/pifm.c
 	$(CC) $(CFLAGS_Pifm) -o ../pifm ../fm/pifm.c  $(LDFLAGS_Pifm)
 
 CFLAGS_Piam	= -Wall -g -O2 -Wno-unused-variable
-LDFLAGS_Piam	= librpitx/src/librpitx.a -lm -lrt -lpthread -lsndfile -L/opt/vc/lib -lbcm_host
+LDFLAGS_Piam	= -lm -lrt -lpthread -lsndfile -L/opt/vc/lib -lrpitx
 ../piam : ../am/piam.c
 	$(CC) $(CFLAGS_Piam) -o ../piam ../am/piam.c  $(LDFLAGS_Piam)
 
 CFLAGS_Pidcf77	= -Wall -g -O2 -Wno-unused-variable
-LDFLAGS_Pidcf77	= librpitx/src/librpitx.a -lm -lrt -lpthread -L/opt/vc/lib -lbcm_host
+LDFLAGS_Pidcf77	= -lm -lrt -lpthread -L/opt/vc/lib -lrpitx
 ../pidcf77 : ../dcf77/pidcf77.c
 	$(CC) $(CFLAGS_Piam) -o ../pidcf77 ../dcf77/pidcf77.c  $(LDFLAGS_Piam)
 clean:
diff --git a/src/chirp/chirp.cpp b/./src/chirp/chirp.cpp
index f6ff2b8..f2f145b 100644
--- a/src/chirp/chirp.cpp
+++ b/./src/chirp/chirp.cpp
@@ -1,5 +1,5 @@
 #include <unistd.h>
-#include "../librpitx/src/librpitx.h"
+#include "librpitx.h"
 #include <unistd.h>
 #include "stdio.h"
 #include <cstring>
diff --git a/src/corel8/corel8.cpp b/./src/corel8/corel8.cpp
index 4a47ba8..1ec1911 100644
--- a/src/corel8/corel8.cpp
+++ b/./src/corel8/corel8.cpp
@@ -14,7 +14,7 @@
 #include <fcntl.h>
 #include <sys/mman.h>
 
-#include "../librpitx/src/librpitx.h"
+#include "librpitx.h"
 #include "costas8.h"
 
 float frequency=14.07e6;
diff --git a/src/dvb/dvbrf.cpp b/./src/dvb/dvbrf.cpp
index 2ac5984..7c3e6a2 100644
--- a/src/dvb/dvbrf.cpp
+++ b/./src/dvb/dvbrf.cpp
@@ -1,5 +1,5 @@
 #include <unistd.h>
-#include "../librpitx/src/librpitx.h"
+#include "librpitx.h"
 #include "stdio.h"
 #include <cstring>
 #include <signal.h>
diff --git a/src/foxhunt/foxhunt.cpp b/./src/foxhunt/foxhunt.cpp
index 78fb2ec..6aaa597 100644
--- a/src/foxhunt/foxhunt.cpp
+++ b/./src/foxhunt/foxhunt.cpp
@@ -14,7 +14,7 @@
 #include <sys/mman.h>
 
 
-#include "../librpitx/src/librpitx.h"
+#include "librpitx.h"
 
 
 int FileFreqTiming;
diff --git a/src/freedv/freedv.cpp b/./src/freedv/freedv.cpp
index 8375ffe..3777596 100644
--- a/src/freedv/freedv.cpp
+++ b/./src/freedv/freedv.cpp
@@ -14,7 +14,7 @@
 #include <fcntl.h>
 #include <sys/mman.h>
 
-#include "../librpitx/src/librpitx.h"
+#include "librpitx.h"
 
 int FileVCO;
 bool running=true;
diff --git a/src/fsq/pifsq.cpp b/./src/fsq/pifsq.cpp
index f55b638..3d60821 100644
--- a/src/fsq/pifsq.cpp
+++ b/./src/fsq/pifsq.cpp
@@ -40,7 +40,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <sys/mman.h>
-#include "../librpitx/src/librpitx.h"
+#include "librpitx.h"
 
 #define TONE_SPACING            8789           // ~8.7890625 Hz
 #define BAUD_2                  7812          // CTC value for 2 baud
diff --git a/src/morse/morse.cpp b/./src/morse/morse.cpp
index f5d9a3a..e6d3db2 100644
--- a/src/morse/morse.cpp
+++ b/./src/morse/morse.cpp
@@ -13,7 +13,7 @@
 #include <fcntl.h>
 #include <sys/mman.h>
 #include <ctype.h>
-#include "../librpitx/src/librpitx.h"
+#include "librpitx.h"
 
 #define MORSECODES 37
 
diff --git a/src/ook/sendook.cpp b/./src/ook/sendook.cpp
index 20a1c91..c891bf8 100644
--- a/src/ook/sendook.cpp
+++ b/./src/ook/sendook.cpp
@@ -1,5 +1,5 @@
 #include <unistd.h>
-#include "../librpitx/src/librpitx.h"
+#include "librpitx.h"
 #include "optparse.h"
 #include <unistd.h>
 #include "stdio.h"
diff --git a/src/opera/opera.cpp b/./src/opera/opera.cpp
index acd8302..4e5bd62 100644
--- a/src/opera/opera.cpp
+++ b/./src/opera/opera.cpp
@@ -37,7 +37,7 @@
 #include "math.h"
 #include <signal.h>
  #include <unistd.h>
-#include "../librpitx/src/librpitx.h"
+#include "librpitx.h"
 bool running=true;
 //#define  __VCpp__  TRUE
 
diff --git a/src/pifmrds/pi_fm_rds.cpp b/./src/pifmrds/pi_fm_rds.cpp
index 19f0fd5..a273dce 100644
--- a/src/pifmrds/pi_fm_rds.cpp
+++ b/./src/pifmrds/pi_fm_rds.cpp
@@ -112,7 +112,7 @@ extern "C"
 #include "control_pipe.h"
 }
 
-#include "../librpitx/src/librpitx.h"
+#include "librpitx.h"
 
 ngfmdmasync *fmmod;
 // The deviation specifies how wide the signal is. 
diff --git a/src/pift8/Makefile b/src/pift8/Makefile
index de96446..a862ecc 100644
--- a/src/pift8/Makefile
+++ b/src/pift8/Makefile
@@ -1,17 +1,17 @@
-CXXFLAGS = -std=c++14 -I.
+CXXFLAGS ?= -std=c++14 -I.
 LDFLAGS = -lm
 
-gen_ft8: gen_ft8.o ft8/encode.o ft8/pack.o ft8/text.o ft8/pack_77.o ft8/encode_91.o common/wave.o
-	$(CXX) $(LDFLAGS) -o $@ $^
+gen_ft8: gen_ft8.o common/wave.o
+	$(CXX) $(LDFLAGS) -o $@ $^ -lpift8
 
-pift8: pift8.o ft8/encode.o ft8/pack.o ft8/text.o ft8/pack_77.o ft8/encode_91.o common/wave.o ../librpitx/src/librpitx.a
-	$(CXX) $(LDFLAGS) -o $@ $^
+pift8: pift8.o common/wave.o
+	$(CXX) $(LDFLAGS) -o $@ $^ -lrpitx -lpift8
 
 .PHONY: run_tests
 
 run_tests: test
 	@./test
 
-test:  test.o ft8/encode.o ft8/pack.o ft8/text.o ft8/pack_77.o ft8/encode_91.o ft8/unpack.o
-	$(CXX) $(LDFLAGS) -o $@ $^
+test:  test.o
+	$(CXX) $(LDFLAGS) -o $@ $^ -lft8
 
diff --git a/src/pift8/pift8.cpp b/./src/pift8/pift8.cpp
index 4ce6fca..23e5ed4 100644
--- a/src/pift8/pift8.cpp
+++ b/./src/pift8/pift8.cpp
@@ -5,12 +5,12 @@
 #include <signal.h>
 #include <unistd.h>
 
-#include "ft8_lib/common/wave.h"
-#include "ft8_lib/ft8/pack.h"
-#include "ft8_lib/ft8/encode.h"
-#include "ft8_lib/ft8/constants.h"
+#include "ft8/wave.h"
+#include "ft8/pack.h"
+#include "ft8/encode.h"
+#include "ft8/constants.h"
 
-#include "../librpitx/src/librpitx.h"
+#include "librpitx.h"
 
 bool running=true;
 
diff --git a/src/pocsag/pocsag.cpp b/./src/pocsag/pocsag.cpp
index c3dc487..e5d80fa 100644
--- a/src/pocsag/pocsag.cpp
+++ b/./src/pocsag/pocsag.cpp
@@ -33,7 +33,7 @@ Fork and modification for rpitx (c)(F5OEO 2018)
 #include <strings.h>
 #include <time.h>
 #include <unistd.h>
-#include "../librpitx/src/librpitx.h"
+#include "librpitx.h"
 
 #define PROGRAM_VERSION "0.3"
 //Check out main() at the bottom of the file
diff --git a/src/rpitxv1/rpitx.cpp b/src/rpitxv1/rpitx.cpp
index 0419c26..7ccf723 100644
--- a/src/rpitxv1/rpitx.cpp
+++ b/src/rpitxv1/rpitx.cpp
@@ -20,7 +20,7 @@
    Helped by a code fragment by PE1NNZ (http://pe1nnz.nl.eu.org/2013/05/direct-ssb-generation-on-pll.html)
  */
 #include <unistd.h>
-#include "../librpitx/src/librpitx.h"
+#include "librpitx.h"
 #include "stdio.h"
 #include <stdarg.h>     /* va_list, va_start, va_arg, va_end */
 #include <cstring>
diff --git a/src/sendiq.cpp b/src/sendiq.cpp
index 1bf1101..878dd2b 100644
--- a/src/sendiq.cpp
+++ b/src/sendiq.cpp
@@ -1,5 +1,5 @@
 #include <unistd.h>
-#include "librpitx/src/librpitx.h"
+#include "librpitx.h"
 #include "stdio.h"
 #include <cstring>
 #include <signal.h>
diff --git a/src/spectrumpaint/spectrum.cpp b/src/spectrumpaint/spectrum.cpp
index c12cc78..383ecc1 100644
--- a/src/spectrumpaint/spectrum.cpp
+++ b/src/spectrumpaint/spectrum.cpp
@@ -14,7 +14,7 @@
 #include <fcntl.h>
 #include <sys/mman.h>
 
-#include "../librpitx/src/librpitx.h"
+#include "librpitx.h"
 
 int FilePicture;
 int FileFreqTiming;
diff --git a/src/sstv/pisstv.cpp b/src/sstv/pisstv.cpp
index 2b697fb..a27e5df 100644
--- a/src/sstv/pisstv.cpp
+++ b/src/sstv/pisstv.cpp
@@ -15,7 +15,7 @@
 #include <sys/mman.h>
 
 
-#include "../librpitx/src/librpitx.h"
+#include "librpitx.h"
 
 int FilePicture;
 int FileFreqTiming;
diff --git a/src/tune.cpp b/src/tune.cpp
index 344a71c..01eea80 100644
--- a/src/tune.cpp
+++ b/src/tune.cpp
@@ -1,5 +1,5 @@
 #include <unistd.h>
-#include "librpitx/src/librpitx.h"
+#include "librpitx.h"
 #include "stdio.h"
 #include <cstring>
 #include <signal.h>
