Index: buildroot/package/armadillo/Config.in
===================================================================
--- buildroot.orig/package/armadillo/Config.in
+++ buildroot/package/armadillo/Config.in
@@ -1,4 +1,4 @@
-comment "armadillo needs a toolchain w/ C++"
+comment "armadillo needs a toolchain w/ C++, fortran"
 	depends on !BR2_INSTALL_LIBSTDCPP
 	depends on !BR2_powerpc
 	depends on !BR2_m68k_cf
@@ -10,9 +10,9 @@ comment "armadillo needs a glibc toolcha
 config BR2_PACKAGE_ARMADILLO
 	bool "armadillo"
 	depends on BR2_INSTALL_LIBSTDCPP
-	depends on !BR2_powerpc || BR2_TOOLCHAIN_USES_GLIBC # clapack
-	depends on !BR2_m68k_cf # clapack
-	select BR2_PACKAGE_CLAPACK
+	depends on BR2_TOOLCHAIN_HAS_FORTRAN
+	depends on !BR2_powerpc || BR2_TOOLCHAIN_USES_GLIBC # lapack
+	select BR2_PACKAGE_LAPACK
 	help
 	  Armadillo: An Open Source C++ Linear Algebra Library for
 	  Fast Prototyping and Computationally Intensive Experiments.
Index: buildroot/package/armadillo/armadillo.mk
===================================================================
--- buildroot.orig/package/armadillo/armadillo.mk
+++ buildroot/package/armadillo/armadillo.mk
@@ -7,7 +7,7 @@
 ARMADILLO_VERSION = 9.900.2
 ARMADILLO_SOURCE = armadillo-$(ARMADILLO_VERSION).tar.xz
 ARMADILLO_SITE = https://downloads.sourceforge.net/project/arma
-ARMADILLO_DEPENDENCIES = clapack
+ARMADILLO_DEPENDENCIES = lapack
 ARMADILLO_INSTALL_STAGING = YES
 ARMADILLO_LICENSE = Apache-2.0
 ARMADILLO_LICENSE_FILES = LICENSE.txt
Index: buildroot/package/gnuradio/gnuradio.mk
===================================================================
--- buildroot.orig/package/gnuradio/gnuradio.mk
+++ buildroot/package/gnuradio/gnuradio.mk
@@ -149,6 +149,13 @@ else
 GNURADIO_CONF_OPTS += -DENABLE_GR_TRELLIS=OFF
 endif
 
+ifeq ($(BR2_PACKAGE_GNURADIO_UHD),y)
+GNURADIO_DEPENDENCIES += uhd
+GNURADIO_CONF_OPTS += -DENABLE_GR_UHD=ON
+else
+GNURADIO_CONF_OPTS += -DENABLE_GR_UHD=OFF
+endif
+
 ifeq ($(BR2_PACKAGE_GNURADIO_UTILS),y)
 GNURADIO_CONF_OPTS += -DENABLE_GR_UTILS=ON
 else
Index: buildroot/package/gnuradio/Config.in
===================================================================
--- buildroot.orig/package/gnuradio/Config.in
+++ buildroot/package/gnuradio/Config.in
@@ -154,4 +154,10 @@ config BR2_PACKAGE_GNURADIO_TRELLIS
 	help
 	  Trellis coded modulation blocks
 
+config BR2_PACKAGE_GNURADIO_UHD
+	bool "gr-uhd support"
+	select BR2_PACKAGE_UHD
+	help
+	  libUHD support
+
 endif
