From b446a046f93f5f2d78085243cf8af2702f260b98 Mon Sep 17 00:00:00 2001
From: Gwenhael Goavec-Merou <gwenhael.goavec-merou@trabucayre.com>
Date: Thu, 26 Oct 2017 15:40:33 +0200
Subject: [PATCH] suppress unitest dependency

Signed-off-by: Gwenhael Goavec-Merou <gwenhael.goavec-merou@trabucayre.com>
---
 host/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/host/CMakeLists.txt b/host/CMakeLists.txt
index 282a580f..4b49daa2 100644
--- a/host/CMakeLists.txt
+++ b/host/CMakeLists.txt
@@ -247,7 +247,6 @@ SET(BOOST_REQUIRED_COMPONENTS
     program_options
     regex
     system
-    unit_test_framework
     serialization
 )
 IF(MINGW)
-- 
2.13.6

