Index: gr-iio-d6d1f6ac7db5be3eb990ee8ef78bb65cf3a15dd7.next/lib/attr_sink_impl.cc
===================================================================
--- gr-iio-d6d1f6ac7db5be3eb990ee8ef78bb65cf3a15dd7.next.orig/lib/attr_sink_impl.cc
+++ gr-iio-d6d1f6ac7db5be3eb990ee8ef78bb65cf3a15dd7.next/lib/attr_sink_impl.cc
@@ -89,7 +89,7 @@ namespace gr {
       }
 
       message_port_register_in(pmt::mp("attr"));
-      set_msg_handler(pmt::mp("attr"), boost::bind(&attr_sink_impl::write_attribute, this, _1));
+      set_msg_handler(pmt::mp("attr"), boost::bind(&attr_sink_impl::write_attribute, this, boost::placeholders::_1));
 
     }
 
